src/dataregistry/registrar/dataset.py

```diff
@@ -924,112 +935,28 @@ def _delete_by_id(self, dataset_id, confirm=False):
 
         self.db_connection.logger.info(f"Deleted {dataset_id} from data registry")
 
-    def _validate_keywords(self, keywords):
+    def add_keywords(self, dataset_id, keyword):
         """
-        Validate a list of keywords.
-
-            - Ensure they are strings
-            - Ensure the chosen keywords are registered in the keywords table
-
-        If any keyword is invalid an exception is raised.
-
-        Parameters
-        ----------
-        keywords : list[str]
-
-        Returns
-        -------
-        keyword_ids : list[int]
-            The associated `keyword_id`s from the `keyword` table
-        """
-
-        keyword_ids = []
-
-        for k in keywords:
-            # Make sure keyword is a string
-            if not isinstance(k, str):
-                raise ValueError(f"{k} is not a valid keyword string")
-
-        # Make sure keywords are all in the keywords table
-        keyword_table = self._get_table_metadata("keyword")
-
-        stmt = select(keyword_table.c.keyword_id).where(
-            keyword_table.c.keyword.in_([x.lower() for x in keywords])
-        )
-
-        with self._engine.connect() as conn:
-            result = conn.execute(stmt)
-            conn.commit()
-
-        # Keyword found
-        for r in result:
-            keyword_ids.append(r.keyword_id)
-
-        # Keyword not found
-        if len(keyword_ids) != len(keywords):
-            raise ValueError("Not all keywords selected are registered")
-
-        return keyword_ids
-
-    def add_keywords(self, dataset_id, keywords):
-        """
-        Add/append keywords to an already existing dataset.
-
-        First check the keywords are valid, then append. If the dataset already
-        has one or more of the passed keywords attributed to it, the keyword(s)
-        will not be duplicated.
+        Add keyword tags to a dataset entry.
 
         Parameters
         ----------
         dataset_id : int
-        keywords : list[str]
+            Dataset to add keyword to
+        keyword : list[str]
+            Keyword to add to dataset
         """
+        self.keyword_table.add_dataset_keywords_relation(dataset_id, keyword)
```

Comment:
Would prefer a different name for the function.   I'll leave a comment about this for keyword.py, where it's defined.